### PR TITLE
Add table descriptor and schema registry usage for Kafka connector

### PR DESCRIPTION
**Pull request by mosabua**
Replaces the stale progress on https://github.com/trinodb/trino/pull/6791 

It would be great to get this merged before 353 @kokosing

### CODE REVIEW

**Comment by yuhuidavidzhang** [comment]
Update Docs

**Review by kokosing — APPROVED** [review]

File: docs/src/main/sphinx/connector/kafka.rst
Comment (elonazoulay) [inline docs/src/main/sphinx/connector/kafka.rst:70]: Can you include the kafka.table-description-supplier config?
```
``kafka.table-description-supplier``                       The table description supplier to use
```

Comment (mosabua) [reply docs/src/main/sphinx/connector/kafka.rst:70]: No .. it is documented separately and the sentence after the table links to all that config

File: docs/src/main/sphinx/connector/kafka.rst
Comment (mosabua) [inline docs/src/main/sphinx/connector/kafka.rst:163]: see here @elonazoulay

Comment (elonazoulay) [reply docs/src/main/sphinx/connector/kafka.rst:163]: LGTM! As long as users can easily see that they have to set that config.

**Comment by mosabua** [comment]
@kokosing or anybody else among  @trinodb/maintainers  ready to merge?

**Comment by kokosing** [comment]
CI hit https://github.com/trinodb/trino/issues/5758

**Comment by kokosing** [comment]
> @kokosing or anybody else among @trinodb/maintainers ready to merge?

I am going to merge it. In case there are more comments we can always apply them as separate pull request.

**Pull request merged**
